independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializerTest.java

```diff
@@ -0,0 +1,138 @@
+    private static Mirror mirrorA;
+    private static Proxy localProxy;
+    private static Settings baseSettings;
+
+    @BeforeClass
+    public static void init() {
+        baseSettings = new Settings();
+
+        baseSettings.setInteractiveMode(true);
+        baseSettings.setUsePluginRegistry(false);
+        baseSettings.setOffline(false);
+
+        final Profile profile = new Profile();
+        profile.setId("foo-profile");
+
+        final RepositoryPolicy releasesPolicy = new RepositoryPolicy();
+        releasesPolicy.setEnabled(true);
+        releasesPolicy.setUpdatePolicy(null);
+        releasesPolicy.setChecksumPolicy(CHECKSUM_POLICY_WARN);
+
+        final RepositoryPolicy snapshotsPolicy = new RepositoryPolicy();
+        snapshotsPolicy.setEnabled(true);
+        snapshotsPolicy.setUpdatePolicy(UPDATE_POLICY_DAILY);
+        snapshotsPolicy.setChecksumPolicy(CHECKSUM_POLICY_WARN);
+
+        final RepositoryPolicy noSnapshotsPolicy = new RepositoryPolicy();
+        noSnapshotsPolicy.setEnabled(false);
+
+        final Repository customRepository = new Repository();
+        customRepository.setId("custom-repo");
+        customRepository.setUrl("https://foo.repo/artifact/content/groups/foo/");
+        customRepository.setReleases(releasesPolicy);
+        customRepository.setSnapshots(snapshotsPolicy);
+        profile.addRepository(customRepository);
+
+        final Repository jbossRepository = new Repository();
+        jbossRepository.setId("jboss-public-repository");
+        jbossRepository.setUrl("https://repository.jboss.org/nexus/content/repositories/releases/");
+        jbossRepository.setReleases(releasesPolicy);
+        jbossRepository.setSnapshots(noSnapshotsPolicy);
+        profile.addRepository(jbossRepository);
+
+        final Repository springRepository = new Repository();
+        springRepository.setId("spring-public-repository");
+        springRepository.setUrl("http://repo.spring.io/libs-release/");
+        springRepository.setReleases(releasesPolicy);
+        springRepository.setSnapshots(noSnapshotsPolicy);
+        profile.addRepository(springRepository);
+
+        baseSettings.addProfile(profile);
+        baseSettings.addActiveProfile("foo-profile");
+
+        localProxy = new Proxy();
+        localProxy.setActive(true);
+        localProxy.setProtocol("http");
+        localProxy.setUsername(null);
+        localProxy.setPassword(null);
+        localProxy.setPort(8888);
+        localProxy.setHost("localhost");
+        localProxy.setNonProxyHosts("localhost");
+        localProxy.setId("local-proxy-http");
+
+        mirrorA = new Mirror();
+        mirrorA.setMirrorOf("central,jboss-public-repository,spring-public-repository");
+        mirrorA.setUrl("https://mirror.com/artifact/content/groups/public/");
+        mirrorA.setId("mirror-A");
+    }
+
+    @Test
+    public void get_remote_repo_from_settings_without_proxy_neither_mmirror() throws AppModelResolverException {

```

Comment:
We are using camelCase for method names all over, let's stick to that. Also let's rephrase it a bit, e.g. `getRemoteRepoFromSettingsWithNeitherProxyNorMirror`